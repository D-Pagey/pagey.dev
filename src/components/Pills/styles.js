import styled from 'styled-components/macro';
import { coloursByFunction } from '../../tokens';

export const List = styled.ul`
    display: flex;
    flex-wrap: wrap;
    list-style: none;
    padding: 0;
`;

export const Pill = styled.li`
    background-color: ${coloursByFunction.pill};
    border-radius: 5px;
    font-size: 12px;
    padding: 0.25rem 0.5rem;
    margin: 0 0 0.5rem 0.5rem;
`;
