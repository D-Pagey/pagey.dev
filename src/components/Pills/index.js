import React from 'react';
import { string } from 'prop-types';
import * as S from './styles';

const Pills = ({ items }) => {
    return (
        <S.List>
            {items.map((item) => (
                <S.Pill key={item}>{item}</S.Pill>
            ))}
        </S.List>
    );
};

Pills.propTypes = {
    items: string.isRequired
};

export default Pills;
