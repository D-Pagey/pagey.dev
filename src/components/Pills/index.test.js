import React from 'react';
import Pills from '.';

const props = {
    items: ['react', 'react-context', 'jest', 'react-testing-library']
};

describe('Pill component', () => {
    it('should render', () => {
        const { container } = render(<Pills {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
