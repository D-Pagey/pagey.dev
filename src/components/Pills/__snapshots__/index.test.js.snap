// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill component should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
  padding: 0;
}

.c1 {
  background-color: #fbce63;
  border-radius: 5px;
  font-size: 12px;
  padding: 0.25rem 0.5rem;
  margin: 0 0 0.5rem 0.5rem;
}

<ul
  class="c0"
>
  <li
    class="c1"
  >
    react
  </li>
  <li
    class="c1"
  >
    react-context
  </li>
  <li
    class="c1"
  >
    jest
  </li>
  <li
    class="c1"
  >
    react-testing-library
  </li>
</ul>
`;
